### PR TITLE
Allow adding `privateEndpoints` to subnets.

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/wepa/issues/55

### Please check if PR meets these requirements

- [ ] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Comment by whites11** [comment]
Using this user config for the `azure-cluster`:

```
---
apiVersion: v1
data:
  values: |
  ...
    connectivity:
      network:
        workers:
          privateEndpoints:
            - name: "whites"
              privateLinkServiceConnections:
                - /subscriptions/6b1f6e4a-6d0e-4aa4-9a5a-fbaca65a23b3/resourceGroups/whites-private-stuff/providers/Microsoft.Network/privateLinkServices/whites
...
kind: ConfigMap
metadata:
  creationTimestamp: null
  labels:
    giantswarm.io/cluster: whites14
  name: whites14-userconfig
  namespace: org-giantswarm
  ```

I got a private endpoint successfully created.

Screenshot_20240221_145948

File: helm/cluster-azure/templates/_helpers.tpl [hunk 1]
```diff
@@ -326,6 +326,23 @@ node-natgateway
 {{- end -}}
 {{- end -}}
 
+{{- define "network.subnet.privateEndpoints" -}}
+{{- if (gt (len $) 0) -}}
```
Comment (AndiDog) [inline helm/cluster-azure/templates/_helpers.tpl:330]:
```suggestion
{{- if (gt (len $.endpoints) 0) -}}
```

And just to be sure, please test that without specifying the new value, the output manifest doesn't change with this PR.

**Review by calvix — APPROVED** [review]

**Review by AndiDog — COMMENTED** [review]
LGTM except one comment

**Comment by whites11** [comment]
cluster-test-suites

**Review by AndiDog — APPROVED** [review]

**Pull request merged**